app/models/grda_warehouse/service_history_service.rb

```diff
@@ -27,47 +28,45 @@ class GrdaWarehouse::ServiceHistoryService < GrdaWarehouseBase
     in_project_type(GrdaWarehouse::Hud::Project::HOMELESS_UNSHELTERED_PROJECT_TYPES)
   end
 
-  scope :in_project_type, -> (project_types) do
-    where(project_type_column => project_types)
-  end
-
-
   scope :homeless_only, -> (start_date:, end_date:) do
-    # CHRONIC_PROJECT_TYPES
-    # HOMELESS_PROJECT_TYPES
-    homeless_project_types = GrdaWarehouse::Hud::Project::HOMELESS_PROJECT_TYPES
-    non_homeless = GrdaWarehouse::Hud::Project::RESIDENTIAL_PROJECT_TYPE_IDS - homeless_project_types
+    to_exclude = service_within_date_range(start_date: start_date, end_date: end_date).
+      ph_moved_in.
+      pluck(:client_id, :date)
 
     where(
-      project_type: homeless_project_types,
-      date: (start_date..end_date)
+      project_type: GrdaWarehouse::Hud::Project::HOMELESS_PROJECT_TYPES,
+      date: (start_date..end_date),
     ).
-    where(
-      GrdaWarehouse::ServiceHistoryService.
-        where(shs_t[:client_id].eq(arel_table[:client_id])).
-        where(date: (start_date..end_date)).
-        where.not(project_type: non_homeless).
-        exists
+    where.not(id:
+      where(
+        client_id: to_exclude.map(&:first),
```

Comment:
I don't think this is going to do what you are expecting.  This will turn into `where client_id in (1, 2, 3) and date in ('2018-01-01', '2018-02-01', '2018-04-01')` which will negate any client on any of those dates.

Reply:
Probably these will need to be joined with ` OR `, but the pluck alone may be too expensive against this table in production.